### PR TITLE
Fix #1107: log unmapped exceptions in addition to mapping

### LINKED ISSUE #1107: Data API exception mapper (`ThrowableToErrorMapper`) does not log error for unmapped server errors (`ErrorCode.SERVER_UNHANDLED_ERROR`)
Currently `ThrowableToErrorMapper` has fallback mapping for unrecognized/unmapped cases:

```
        return ErrorCode.SERVER_UNHANDLED_ERROR
            .toApiException("root cause: (%s) %s", throwable.getClass().getName(), message)
            .getCommandResultError(Response.Status.INTERNAL_SERVER_ERROR);
```

which makes sense. But since this is a failure to map, we want to get rid of all such cases: for this, we'd need to actually log these cases.
So let's add log statement.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds error logic for unmapped exceptions by Data API.

**Which issue(s) this PR fixes**:
Fixes #1107

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java [hunk 1]
```diff
@@ -46,6 +50,10 @@ public final class ThrowableToErrorMapper {
         }
 
         // handle all other exceptions
+        logger.error(
+            String.format(
+                "Unrecognized exception caught, mapped to SERVER_UNHANDLED_ERROR: %s", message),
+            throwable);
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java:56]: Will log, including stack trace, when `Throwable` passed.

**Review by Yuqi-Du — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java (hunk 1 above)
Comment (jeffreyscarpenter) [reply src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java:56]: possible for this to be tied to tenant ID?

Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java:56]: Will try to see: cannot pass directly (one of things I dislike about Dependency Injection style), but perhaps there's a way with context (MDC?)

Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java:56]: Looks like we do set `tenantId` entry of MDC in `MeteredCommandProcessor`. And assuming the new logger uses default formatting, and that is includes contents of MDC under `"mdc"` key (which I _think_) happens, I think it should be automatically included.

**Pull request merged**